### PR TITLE
feature/leetcode-199: Implement Popover component

### LINKED ISSUE #199: Popover — MUI CDD Implementation
## Summary
Implement the `Popover` component following CDD principles using Material UI. The component displays rich floating content anchored to a trigger element and must position itself correctly within the viewport on all screen sizes.

## Type
Task

## Priority
Medium

## Labels
- CDD
- MUI
- Component
- Frontend
- Responsive

---

## Figma Reference
[View Figma Design](https://www.figma.com/design/1EYOnbPan3uCJFjlUbeGXc/Untitled?node-id=0-1&t=sSJuoUv8FEzXRTvJ-1)

---

## Description
Create a `Popover` component built on top of MUI's `Popover` primitive. It must support configurable placement (`top`, `bottom`, `left`, `right`) with automatic flip when there is insufficient space, arrow indicator, custom content slot, and a close trigger. On mobile it must resize to stay within viewport bounds.

---

## Mandatory File Structure

```
Popover/
├── Popover.tsx              # Main component logic & JSX
├── Popover.types.ts         # Props interfaces (extend MUI PopoverProps)
├── Popover.constants.ts     # Placement enum, arrow size constant (optional)
├── Popover.hook.ts          # Anchor element open/close hook (optional)
├── Popover.styles.ts        # MUI styled() / sx styles — no external CSS, no hardcoded px
├── Popover.utils.ts         # Arrow position calculation helper (optional)
├── Popover.test.tsx         # Unit tests — Jest / Testing Library
├── Popover.stories.tsx      # Storybook stories in CSF format
└── Popover.docs.mdx         # MDX documentation for Storybook
```

---

## Acceptance Criteria
- [ ] Component implemented using Material UI (`@mui/material/Popover`)
- [ ] Configurable `placement` prop: `top`, `bottom`, `left`, `right` (and `-start`/`-end` variants)
- [ ] Auto-flips placement when there is not enough space
- [ ] Optional arrow indicator pointing to the trigger
- [ ] Arbitrary content rendered in the popover body
- [ ] Closes on click-outside and Escape key
- [ ] On mobile (< 600px) renders full-width and vertically centred
- [ ] Fully responsive — tested at 320px, 375px, 768px, 1024px
- [ ] TypeScript types defined in `Popover.types.ts`
- [ ] Styles in `Popover.styles.ts` using `styled()` or `sx` — no external CSS, no hardcoded `px`
- [ ] Unit tests covering open/close, placement, flip, and Escape key
- [ ] Storybook stories in CSF format
- [ ] MDX documentation written in English
- [ ] Passes axe / ARIA accessibility checks (`aria-haspopup`, `aria-expanded`)
- [ ] Figma design fidelity verified

---

## Technical Notes
- Base on `@mui/material/Popover`
- Use `anchorOrigin` / `transformOrigin` for positioning
- Arrow implemented with a rotated `Box` using `sx` — no CSS files
- No hardcoded `px` values — use MUI theme tokens
- All comments and documentation must be written in **English**

### PR DESCRIPTION
**Pull request by rodrigopaco**
Implemented the Popover component following CDD principles using Material UI.

https://ditmar.github.io/leetcode-spa/feature-leetcode-199

Includes:
- MUI Popover-based implementation
- Configurable placement support for top, bottom, left, right, and start/end variants
- Auto-flip behavior when there is not enough viewport space
- Optional arrow indicator
- Arbitrary content rendering through children
- Close behavior on click outside and Escape key
- Responsive mobile layout for viewports below 600px
- ARIA attributes for accessibility
- Unit tests with Vitest and Testing Library
- Storybook CSF stories
- English MDX documentation

### CODE REVIEW

**Review by AlejandroGus-ChJ — APPROVED** [review]
"Hola Rodrigo, he revisado tus archivos. El componente Popover está muy bien estructurado y las interfaces en types.ts definen claramente cada propiedad. Me parece bien que hayas incluido los atributos de accesibilidad (aria-labels) y que todos los checks automáticos estén en verde. ¡Buen trabajo!"

File: src/component-catalog/Popover/Popover.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          role="dialog"
+          data-testid="popover-content"
+          data-placement={activePlacement}
+        >
+          {showArrow && <PopoverArrow arrowSide={arrowSide} data-testid="popover-arrow" />}
+          {children}
+        </PopoverContent>
+      </MuiPopover>
+    </>
+  );
```
Comment (Ditmar) [inline src/component-catalog/Popover/Popover.tsx:77]: The Popover component does not handle the case where the `trigger` prop is not provided. This could lead to runtime errors when trying to render the trigger element.

**Suggestion:**
```typescript
Add a check to ensure that the `trigger` prop is provided and handle the case where it is not, possibly by rendering a fallback or throwing an error.
```

File: src/component-catalog/Popover/Popover.hook.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }, [anchorEl, open, placement]);
+
+  return {
+    anchorEl,
+    open,
+    activePlacement,
+    contentRef,
+    handleOpen,
+    handleClose,
+  };
```
Comment (Ditmar) [inline src/component-catalog/Popover/Popover.hook.ts:75]: The `onOpenChange` callback is called without checking if it is defined. This could lead to runtime errors if the parent component does not provide this prop.

**Suggestion:**
```typescript
Ensure that `onOpenChange` is defined before calling it to avoid potential errors.
```

File: src/component-catalog/Popover/Popover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        toJSON: () => undefined,
+      },
+      popoverWidth: 320,
+      popoverHeight: 200,
+      viewportWidth: 1024,
+      viewportHeight: 300,
+    });
+
+    expect(placement).toBe('top');
+  });
```
Comment (Ditmar) [inline src/component-catalog/Popover/Popover.test.tsx:189]: The test for auto-flipping placement does not cover scenarios where the popover might not have enough space in both directions. This could lead to untested edge cases.

**Suggestion:**
```typescript
Add additional tests to cover scenarios where the popover placement needs to be flipped multiple times or where it might not fit in any direction.
```

**Review by Ditmar — CHANGES_REQUESTED** [review]
The code introduces a Popover component with various features and configurations. Overall, the implementation appears well-structured and follows good practices. However, there are some areas for improvement regarding accessibility and error handling that should be addressed to enhance robustness.
---

**Conclusion:** While the implementation is generally solid, addressing the identified issues related to accessibility and error handling is crucial for ensuring robustness and preventing potential runtime errors.

File: src/component-catalog/Popover/Popover.hook.ts (hunk 2 above)
Comment (rodrigopaco) [reply src/component-catalog/Popover/Popover.hook.ts:75]: Fixed. Updated the hook to explicitly check whether `onOpenChange` is defined before calling it in both open and close handlers, this keeps the callback optional and avoids potential runtime issues.

File: src/component-catalog/Popover/Popover.test.tsx (hunk 3 above)
Comment (rodrigopaco) [reply src/component-catalog/Popover/Popover.test.tsx:189]: Fixed. Expanded the auto-flip unit tests to cover additional edge cases, including when the popover cannot fit in either direction and when horizontal placements need to flip.

File: src/component-catalog/Popover/Popover.tsx (hunk 1 above)
Comment (rodrigopaco) [reply src/component-catalog/Popover/Popover.tsx:77]: Fixed. Added a runtime guard so the component returns null when the trigger prop is not provided. I also added a unit test to verify that the popover does not render when the trigger is missing.

**Review by Ditmar — APPROVED** [review]
